build.gradle

```
@@ -535,6 +541,24 @@ configure(subprojects - getJsProjects()) { subproject ->
     }
 }
 
+def checkDexer = task('checkDexer', description: 'compiles android class files into dex bytecode', dependsOn: compileKotlin) {
+    getAndroidProjects().each { p->
+        doLast {
+            javaexec {
+                def nameWithoutSuffix = p.name.substring(p.name.indexOf('-') + 1, p.name.lastIndexOf('-'))
+                def jvmProject = prefixedProject(nameWithoutSuffix + '-jvm')
+                def jarToTest = "${jvmProject.projectDir}\\build\\libs\\*.jar"
+                mkdir "d8out/atrium-${nameWithoutSuffix}-${rootProject.version}"
+                classpath = configurations.r8
+                main = "com.android.tools.r8.D8"
+                args = ['--output' , "d8out/atrium-${nameWithoutSuffix}-${rootProject.version}", "${jarToTest}"]
```

```suggestion
                args = ['--output' , "d8out/atrium-${nameWithoutSuffix}-${rootProject.version}", p.jar.archiveFile.get().asFile]
```
maybe better but requires that you wrap the whole thing into an `afterEvaluate {..}` block as the archiveFile does not include the version otherwise